tests/unit/plugins/modules/test_redfish_storage_volume.py

```diff
@@ -608,3 +742,60 @@ def test_check_mode_validation(self, redfish_connection_mock_for_storage_volume,
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_success_case03(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID6", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_success_case04(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID60", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_failure_case(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                    redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID9", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        with pytest.raises(Exception) as exc:
+            self.module.check_raid_type_supported(f_module,
+                                                  redfish_connection_mock_for_storage_volume)
+        assert exc.value.args[0] == "RAID Type RAID9 is not supported."
+
+    def test_check_raid_type_supported_exception_case(self, redfish_response_mock,
+                                                      redfish_connection_mock_for_storage_volume,
+                                                      storage_volume_base_uri):
+        param = {"volume_type": "NonRedundant", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_connection_mock_for_storage_volume.invoke_request.side_effect = HTTPError('http://testhost.com', 400,
```

Comment:
use https instead of http, you'll avoid so many sonar lint issue, it just mocking

Reply:
done